core/domain/state_domain.py

```diff
@@ -2070,15 +2073,19 @@ class SubtitledHtml(python_utils.OBJECT):
 
     def __init__(self, content_id, html):
         """Initializes a SubtitledHtml domain object. Note that initializing
-        the SubtitledHtml object does not clean the html. Before saving the
-        SubtitledHtml object, validate should be called for validation and
+        the SubtitledHtml object does not clean the html. This is because we
+        sometimes need to initalize SubtitledHtml and migrate the contained
+        html from an old schema, but the cleaner would remove invalid tags
+        and attributes before having a chance to migrate it. Before saving the
+        SubtitledHtml object, validate() should be called for validation and
         cleaning of the html.
 
         Args:
             content_id: str. A unique id referring to the other assets for this
                 content.
-            html: str. A piece of user-submitted HTML. This is cleaned in such
-                a way as to contain a restricted set of HTML tags.
+            html: str. A piece of user-submitted HTML. Note that this is NOT
+                cleaned in such a way as to contain a restricted set of HTML
+                tags until the validate() method is called.
```

Comment:
... tags. To clean it, the validate() method must be called.

Reply:
Done